src/stores/NuxSoftwareSetupStore.js

```diff
@@ -0,0 +1,97 @@
+// @flow
+
+import type { Device, Location } from 'brewskey.js-api';
+
+import { action, observable } from 'mobx';
+import NavigationService from '../NavigationService';
+
+type IntermediateData = {
+  location?: Location,
+};
+
+class NuxSoftwareSetupStore {
+  @observable _intermediateData: IntermediateData = {};
+
+  onGetStartedPress = async () => {
+    NavigationService.navigate('nuxLocation', {
+      onContinuePress: (location?: Location) => {
+        if (location) {
+          this._onGetLocation(location);
+        } else {
+          NavigationService.navigate('newLocation', {
+            onLocationCreated: this._onGetLocation,
+            showBackButton: false,
+          });
+        }
+      },
+    });
+  };
+
+  _onGetLocation = (location: Location) => {
+    this._updateIntermediateData({ location });
+    NavigationService.navigate('nuxWifi', {
+      onContinuePress: () => {
+        NavigationService.navigate('wifiSetup', {
+          forNewDevice: true,
+          onSetupFinish: this._onWifiSetupFinish,
+        });
+      },
+    });
+  };
+
+  _onWifiSetupFinish = (particleID: string) => {
+    NavigationService.navigate('nuxDevice', {
+      onContinuePress: () => {
+        const { location } = this._intermediateData;
+
+        NavigationService.navigate('newDevice', {
+          hideLocation: true,
```

Comment:
This field is hidden `only` on nux device creation flow, because we choose a location on the first step of NUX. I don't think that's its great idea to hide it anywhere/anywhen other than that.